### PR TITLE
Solar system activity

### PR DESCRIPTION
**Pull request by gupta2140**
Fresh PR for solar_system activity.

### CODE REVIEW

File: src/activities/solar_system/Solar_system.qml [hunk 1]
```diff
@@ -0,0 +1,321 @@
+/* GCompris - Solar_system.qml
```
Comment (petitlapin) [inline src/activities/solar_system/Solar_system.qml]: need to be renamed to SolarSystem.qml

File: src/activities/solar_system/QuizScreen.qml [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+
+    // This grid has image of the planet in its first column/row (row in case of vertical screen) and the options on the 2nd column/row
+    Grid {
+        id: imageAndOptionGrid
+        columns: (background.horizontalLayout && !items.assessmentMode && items.bar.level != 2) ? 2 : 1
+        spacing: 10 * ApplicationInfo.ratio
+        anchors.top: (questionArea.y + questionArea.height) > (score.y + score.height) ? questionArea.bottom : score.bottom
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.margins: 10 * ApplicationInfo.ratio
```
Comment (petitlapin) [inline src/activities/solar_system/QuizScreen.qml]: can you add topMargin to 0? Else on some ratios, the latest answer overrides the closeness bar

File: src/activities/solar_system/Solar_system.qml [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+            property string lengthOfYearHint
+        }
+
+        onStart: {
+            dialogActivityConfig.getInitialConfiguration()
+            Activity.start(items)
+        }
+
+        onStop: {
+            dialogActivityConfig.saveDatainConfiguration()
```
Comment (petitlapin) [inline src/activities/solar_system/Solar_system.qml]: you don't need to save the configuration here, it's already done when changing it

**Comment by gupta2140** [comment]
Regarding clicking on options when the animation is running, that's something that had to be done with AnswerButton.qml and since now we have the blockAllButtonClicks there, it will be used here too.
It can be done when merged, else rebasing the branch will again cause review problems.

File: src/activities/solar_system/Solar_system.qml [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+                top: parent.top
+                topMargin: 10
+                right: parent.right
+                rightMargin: 5
+                left: parent.left
+                leftMargin: 5
+            }
+            z: 10
+
+            readonly property string commonInstruction: qsTr("Mode: <font color=\"#3bb0de\">%1</font><br><br>There are two modes in the activity which you can switch from the configuration window:<br><b>1. Normal mode</b> - In this mode you can play and learn about the Solar System.<br><b>2. Assessment mode</b> - In this mode you can test your knowledge about the Solar System.").arg(items.assessmentMode ? "Assessment" : "Normal")
```
Comment (petitlapin) [inline src/activities/solar_system/Solar_system.qml]: missing qsTr for Assessment and Normal

**Comment by petitlapin** [comment]
Merged in https://commits.kde.org/gcompris/0c3069be9ddaae7531624e10c4ff4dd16f0e79f6

Thank you

**Pull request closed without merging**